src/BloomExe/HtmlThumbNailer.cs

```diff
@@ -98,6 +98,56 @@ public enum BorderStyles
 
 
 		/// <summary>
+		/// A synchronous version of getting thumbnails, currently used by the image server
+		/// to get thumbnails that are used in the add page dialog.
+		/// </summary>
+		/// <param name="key">Used to retrieve the thumbnail from a dictionary if we are asked for
+		/// the same one repeatedly</param>
+		/// <param name="document">Whose rendering will produce the thumbnail content.</param>
+		/// <param name="options"></param>
+		/// <returns></returns>
```

Comment:
We could use some kind of convention to tell whether the caller should dispose of the result. But lacking that, maybe a comment here saying that the result is cached so the caller should not dispose of it.